www/README.md

```diff
@@ -0,0 +1,121 @@
+
+### Serving the app during development
+
+The app can be served through `kubectl`, but for some types of review a local web server is convenient. One can be installed as follows:
+
+```
+sudo npm install -g http-server
+```
+
+The server can then be launched:
+
+```
+cd app
+http-server -a localhost -p 8000
+```
+
+### Configuration
+#### Configuration settings
+A json file can be used by `gulp` to automatically create angular constants. This is useful for setting per environment variables such as api endpoints.
+*  ```www/master/shared/config/development.json``` or ```www/master/shared/config/production.json``` can be created from the ```www/master/shared/config/development.example.json``` file.
+* ```development.example.json``` should be kept up to date with default values, since ```development.json``` is not under source control.
+* Component configuration can be added to ```www/master/components/<component name>/config/development.json``` and it will be combined with the main app config files and compiled into the intermediary ```www/master/shared/config/generated-config.js``` file.
+* All ```generated-config.js``` is compiled into ```app.js```
+* Production config can be generated using ```gulp config --env production``` or ```gulp --env production```
+* The generated angular constant is named ```ENV``` with the shared root and each component having their own child configuration. For example,
+```
+www/master
+├── shared/config/development.json
+└── components
+    ├── dashboard/config/development.json
+    ├── graph/config/development.json
+    └── my_component/config/development.json
+```
+produces ```www/master/shared/config/generated-config.js```:
+```
+angular.module('kubernetesApp.config', [])
+.constant('ENV', {
+  '/': <www/master/shared/config/development.json>,
+  'dashboard': <www/master/components/dashboard/config/development.json>,
+  'graph': <www/master/components/graph/config/development.json>,
+  'my_component': <www/master/components/my_component/config/development.json>
+});
+```
+
+#### Kubernetes server configuration
+
+**RECOMMENDED**: By default the Kubernetes api server does not support CORS,
+  so the `kube-apiserver.service` must be started with
+  `--cors_allowed_origins=.*` or `--cors_allowed_origins=http://<your
+  host here>`
+
+**HACKS**: If you don't want to/cannot restart the Kubernetes api server:
+* Or you can start your browser with web security disabled. For
+  Chrome, you can [launch](http://www.chromium.org/developers/how-tos/run-chromium-with-flags) it with flag ```--disable-web-security```.
+
+### Building a new visualizer or component
+
+See [master/components/README.md](master/components/README.md).
+
+### Testing
+Currently kuberntes-ui includes both unit-testing (run via [Karma](http://karma-runner.github.io/0.12/index.html)) and
+end-to-end testing (run via
+[Protractor](http://angular.github.io/protractor/#/)).
+
+#### Unittests via Karma
+To run the existing Karma tests:
```

Comment:
If you want these tests to remain passing for more than three days, we need travis/shippable to run them automatically. Which means you'll have to add this setup to the .travis.yml file. We can do this in a future PR.